### DISCUSSION TITLE
Fish shell: Alt + ← (and Alt + →) doesn't navigate through directory history

### DISCUSSION THREAD

**Discussion by mattias-arrelid** [ghostty-org/ghostty#3681] [Feature Requests, Ideas] [answered]
By default, fish lets you navigate the directory history using Alt + ←/→ (if you didn't type anything into your prompt):
https://fishshell.com/docs/current/interactive.html#navigating-directories

ghostty binds alt+left (and alt+right) by default:
```
❱ ghostty +list-keybinds | rg left
keybind = super+shift+left_bracket=previous_tab
keybind = shift+left=adjust_selection:left
keybind = super+ctrl+left=resize_split:left,10
keybind = alt+left=esc:b
keybind = super+left=text:\x01
keybind = super+alt+left=goto_split:left
keybind = super+left_bracket=goto_split:previous
```

If one unbinds alt+left and alt+right in the config, directory navigation, as well as jumping by word (back/forward), works just fine again:
```
keybind = alt+left=unbind
keybind = alt+right=unbind
```

Is this expected? The [shell integration documentation](https://ghostty.org/docs/features/shell-integration) doesn't mention shell specific features (like this one), so maybe it's expected that you'd need to override (unbind) in this case.

My environment:
- OS: macOS 15.2 (24C101)
- fish: 3.7.1 (installed using `brew`, ghostty shell integration seems to be working)
- ghosty: 1.0.0 ([8469](https://github.com/ghostty-org/ghostty/commits/4b4d4062))

**Comment by 00-kat** [comment, marked as answer]
https://github.com/ghostty-org/ghostty/discussions/3207#discussioncomment-11673995 (tl;dr: yes it's like this to match `Terminal.app` behaviour which lots of macOS users expect)

**Reply by krobelus** [reply]
> (tl;dr: yes it's like this to match Terminal.app behaviour which lots of macOS users expect)

How can one detect if a terminal has this behavior? i.e. `option-left` sends `alt-b`
I would prefer to not hardcode the name of each terminal that has this feature.
Besides ghostty, it seems to affect only Temrinal.app, according to [the workarounds used in fish](https://github.com/fish-shell/fish-shell/issues/11004#issuecomment-2571478965)

(FWIW I don't see why sending `alt-b` is a good idea, it means that applications everywhere have to work around this: tmux/fish/vim and other editors etc.)

**Reply by 00-kat** [reply]
I don't think it's possible to detect this behaviour without hacky solutions :(

> I don't see why sending `alt-b` is a good idea

This was discussed in https://github.com/ghostty-org/ghostty/discussions/2363#discussioncomment-10824805.  (I can't comment more on this since I don't use macOS.)

**Reply by krobelus** [reply]
https://github.com/ghostty-org/ghostty/pull/4612

**Comment by pgib** [comment]
Thank you. I was wondering why only my up and down were working in Tmux with these bindings:

```
bind-key -n M-up select-pane -U
bind-key -n M-down select-pane -D
bind-key -n M-left select-pane -L
bind-key -n M-right select-pane -R
```

**Comment by mattias-arrelid** [comment]
Fish has implemented a fix for this as of 4.0.6:
https://fishshell.com/docs/current/relnotes.html

> alt-left alt-right were misinterpreted because they send unexpected sequences on some terminals; a workaround has been added
